autoload/lsp.vim

```diff
@@ -943,6 +946,15 @@ function! s:on_request(server_name, id, request) abort
         endif
     elseif a:request['method'] ==# 'window/workDoneProgress/create'
         call s:send_response(a:server_name, { 'id': a:request['id'], 'result': v:null})
+    elseif a:request['method'] ==# 'window/showDocument'
+        if !g:lsp_experimental_show_document | return | endif
+        let l:cmd = g:lsp_show_document_command
+        if has_key(a:request['params'], 'selection')
+            let l:cmd .= ' +' . a:request['params']['selection']['start']['line']
+        endif
+        let l:cmd .= ' ' . lsp#utils#uri_to_path(a:request['params']['uri'])
+        execute l:cmd
+        call s:send_response(a:server_name, { 'id': a:request['id'], 'result': {'success': v:true}})
```

Comment:
This will not work for multibytes chars. I would suggest using utils from the following instead.

* https://github.com/prabirshrestha/vim-lsp/blob/master/autoload/lsp/utils/location.vim
* https://github.com/prabirshrestha/vim-lsp/blob/master/autoload/lsp/utils/range.vim
* https://github.com/prabirshrestha/vim-lsp/blob/master/autoload/lsp/utils/position.vim